### ISSUE TITLE
Finding `dot` fails for conda install of `graphviz` on Windows

### ISSUE THREAD

**Issue by vidartf** [xflr6/graphviz#34]
Installing the `python-graphviz` conda package also fetches the `graphviz` conda package. On Windows, `graphviz` binaries are put into `Library/bin/graphviz`, and `dot.bat` etc. is put in `Library/bin`, which is on PATH. However, `.bat` files are not automatically recognized for execution by `subprocess.Popen`, which causes the engine execution to fail:

```python
RuntimeError: failed to execute ['dot', '-Tsvg'], make sure the Graphviz executables are on your systems' path
```

A specific fix for this could be to try to launch the engine with a `.bat` postfix if Popen fails on Windows.

A more general strategy on Windows could be to iterate over all extensions in os.environ['PathExt'] and attempt those.

A third option would be to allow the `engine` argument to be specified with a file extension, i.e. the extension is stripped with `os.path.splitext(engine)[0]` before validating against `ENGINES`. This would place more of a burden on the user, but it also gives the user more control.

**Comment by xflr6** [comment]
Thanks. This concerns downstream [conda-forge/python-graphviz](https://anaconda.org/conda-forge/python-graphviz) and the [anaconda Graphviz package](https://anaconda.org/anaconda/graphviz) introducing the `.bat` wrappers (a vanilla Graphviz installation uses plain `.exe` binaries).

Looks like the issue is already known downstream: conda-forge/python-graphviz-feedstock#2
It might be better to fix it there. As for the implementation, it might be better to append `.bat` to the engine command on Windows in that package, or fall back to launch with `.bat` as you proposed (the patch in the pull request uses `shell=True`, which AFAICS may be a security risk).

**Comment by vidartf** [comment]
Thanks for pointing out the downstream issue, I wasn't sure where to best submit it. Regarding security risk of `shell=True`, that should not be an issue as the arguments are validated against a whitelist here, effectively preventing the risk of a shell code-injection.

That said, executables on Windows almost always have an extension, and Popen on Windows says extensions are significant. As such, having this package enforce extension-less engine commands seem counter-intuitive (i.e. this will be an issue on Windows irregardless of whether Graphviz was installed by Conda or not).

**Comment by vidartf** [comment]
Update: I read the docs again, and Popen on Windows ([CreateProcess](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682425(v=vs.85).aspx)) does imply `.exe` if the extension is missing, so it might be acceptable to not include an extension for the engine.

**Comment by xflr6** [comment]
Yes, this machine here is actually running Windows, so I'm sure it works with vanilla Graphviz. :)

Right, the executables are whitelisted, but I imagine that a user-provided filename could be (ab)used for some kind of shell injection (but I am no security expert).

Added my proposed fix here (so this can be closed): conda-forge/python-graphviz-feedstock#6
